pkg/binaries/fetch/processor.go

```diff
@@ -76,7 +76,7 @@ func (s *Processor) Stager(baseDir string, bufferSize int64, binary state.Binary
 		NewEphemeralStorage(),
 		NewHTTPFetcher(
 			replaceVars(binary.URLPattern, map[string]string{
-				"#{os}":   ExpectedReleaseAssetNamingConvention(s.Host.Os),
+				"#{os}":   s.Host.Os,
```

Comment:
This was the bug. When downloading kubectl, okctl tried to download

https://amazon-eks.s3.us-west-2.amazonaws.com/1.18.9/2020-11-02/bin/Linux/amd64/kubectl

when it should download

https://amazon-eks.s3.us-west-2.amazonaws.com/1.18.9/2020-11-02/bin/linux/amd64/kubectl

(see the lower case `l` in linux).